tests/proto3/proto3_compat_test.go

```diff
@@ -173,8 +183,154 @@ func TestProto3CompatPtrsRoundtrip(t *testing.T) {
 
 	assert.EqualValues(t, p3ToAm, amToP3)
 
-
 	assert.NotZero(t, len(ab), "expected a non-empty encoding for a non-nil pointer to an empty struct")
 	t.Log(ab)
 
 }
+
+// ---------------------------------------------------------------
+//  ---- time.Time <-> timestamp.Timestamp (proto3 well known type) :
+// ---------------------------------------------------------------
+
+// equivalent go struct or "type" to the proto3 message:
+type goAminoGotTime struct {
+	T *time.Time
+}
+
+func TestProto3CompatEmptyTimestamp(t *testing.T) {
+	empty := p3.ProtoGotTime{}
+	// protobuf also marshals to empty bytes here:
+	pb, err := proto.Marshal(&empty)
+	assert.NoError(t, err)
+	assert.Len(t, pb, 0)
+
+	// unmarshaling an empty slice behaves a bit differently in proto3 compared to amino:
+	res := &goAminoGotTime{}
+	err = cdc.UnmarshalBinaryBare(pb, res)
+	assert.NoError(t, err)
+	// NOTE: this behaves differently because amino defaults the time to 1970-01-01 00:00:00 +0000 UTC while
+	// decoding; protobuf defaults to nil here (see the following lines below):
+	assert.NoError(t, err)
+	assert.Equal(t, goAminoGotTime{T: &epoch}, *res)
+	pbRes := p3.ProtoGotTime{}
+	err = proto.Unmarshal(pb, &pbRes)
+	assert.NoError(t, err)
+	assert.Equal(t, p3.ProtoGotTime{T: nil}, pbRes)
+}
+
+func TestProto3CompatTimestampNow(t *testing.T) {
```

Comment:
Can this be table driven?

Reply:
That would be nice. But it's basically one test-case that documents what works how.